dash-renderer/src/actions/dependencies.js

```diff
@@ -464,9 +464,17 @@ function wildcardOverlap({id, property}, objs) {
 }
 
 export function validateCallbacksToLayout(state_, dispatchError) {
-    const {config, graphs, layout, paths} = state_;
-    const {outputMap, inputMap, outputPatterns, inputPatterns} = graphs;
+    const {config, graphs, layout: layout_, paths: paths_} = state_;
     const validateIds = !config.suppress_callback_exceptions;
+    let layout, paths;
+    if (validateIds && config.validation_layout) {
+        layout = config.validation_layout;
```

Comment:
Is this because `validation_layout` is pruned down, so faster to crawl?

Reply:
This is just saying *if* we have a separate validation layout, that's the one we need to check the callbacks against. The fact that it's pruned should make `computePaths` (next line) faster, but the main reason for pruning was to reduce the size of `_dash-config` on the index page - I was imagining it could become enormous, if you have multiple tables and graphs on different pages all concatenated together.

Reply:
Ah I see, because we're validating the components & component props in the front-end as well now, not just the callback signatures?

Reply:
Whereas previously we were validating the components's properties in the render cycle I think?